source/server/BUILD

```diff
@@ -250,6 +250,41 @@ envoy_cc_library(
     name = "listener_manager_lib",
     srcs = ["listener_manager_impl.cc"],
     hdrs = ["listener_manager_impl.h"],
+    deps = [
+        ":configuration_lib",
+        ":drain_manager_lib",
+        ":filter_chain_manager_lib",
+        ":lds_api_lib",
+        ":transport_socket_config_lib",
+        "//include/envoy/server:filter_config_interface",
+        "//include/envoy/server:listener_manager_interface",
+        "//include/envoy/server:transport_socket_config_interface",
+        "//include/envoy/server:worker_interface",
+        "//source/common/api:os_sys_calls_lib",
+        "//source/common/common:empty_string",
+        "//source/common/config:utility_lib",
+        "//source/common/init:manager_lib",
+        "//source/common/network:cidr_range_lib",
+        "//source/common/network:lc_trie_lib",
+        "//source/common/network:listen_socket_lib",
+        "//source/common/network:resolver_lib",
+        "//source/common/network:socket_option_factory_lib",
+        "//source/common/network:utility_lib",
+        "//source/common/protobuf:utility_lib",
+        "//source/extensions/filters/listener:well_known_names",
+        "//source/extensions/filters/network:well_known_names",
+        "//source/extensions/transport_sockets:well_known_names",
+        "//source/extensions/transport_sockets/tls:context_config_lib",
+        "//source/extensions/transport_sockets/tls:context_lib",
+        "@envoy_api//envoy/admin/v2alpha:config_dump_cc",
+        "@envoy_api//envoy/api/v2:lds_cc",
+    ],
+)
+
+envoy_cc_library(
+    name = "filter_chain_manager_lib",
+    srcs = ["filter_chain_manager_impl.cc"],
+    hdrs = ["filter_chain_manager_impl.h"],
     deps = [
```

Comment:
I don't think you need all deps from listener_manager_impl, can this be less? same for above.

Reply:
Great catch! I believe the list here is beyond what is needed. 
I will manually remove it. Is there a tool to do that?

Reply:
A clean up by brain. tested by build filter_chain_manager_lib and listener_manager_lib